### DISCUSSION TITLE
Trouble using shouldSkip

### DISCUSSION THREAD

**Discussion by eimantaaas** [spatie/laravel-livewire-wizard#15] [Q&A] [answered]
Method `shouldSkip` seems to be not working as expected.

Problems faced:

1. Does not skip the step even if value is set to `true`
2. Clicking on previous step button after steps that should be skipped stays on the same step
3. Skipped step stays visible in steps navigation

**Comment by freekmurze** [comment]
@ju5t could you take a look at this one?

**Reply by ju5t** [reply]
I will have a look at it tomorrow.

1. I will add a skipped step to the demo app to see if I can replicate it. The test for it pass but it might be something else.
2. That's an oversight on my end. It will always go to the next step. @freekmurze any preference for storing the previous step? This can either be a property on the Wizard itself and that's added to the view or something that gets passed along as `$currentState`.
3. Navigation is up to you. I personally prefer to show all steps. If you don't, you should be able to call `shouldSkip()` in your navigation. Just like you can do `isCurrent()`.

**Comment by ju5t** [comment]
So I eventually spent some time on it this evening.

- shouldSkip does not work because it emits an event before Javascript is loaded. As we do not use Dusk, our current test is limited and did not catch it.
- This means we first have to find a solution for it before support going backwards. `wire:init` could be one, but it's not ideal.
- I was wrong. Right now it's not supported. This needs support in `Step`, which it doesn't have yet.

Ideally Livewire would have something like `$this->nextTick($callable)`, like Alpine and Vue.

**Comment by freekmurze** [comment, marked as answer]
I've removed the functionality, and will bring it back in a better way when I need this myself.

**Reply by ju5t** [reply]
I've opened https://github.com/livewire/livewire/discussions/4982 as I do think this is what we need to get this up and running.

In the meantime, for those who want to skip a step, you should be able to do something like this in the skipped component:

```php
public function render()
{
    return '<div wire:init="nextStep"></div>';
}
```

Instead of pointing it to `nextStep` you can make this dynamic too by pointing it to a different method.

**Reply by eimantaaas** [reply]
This could be as temporary solution, but with this solution it is not possible to go back to previous step after a skipped step

**Reply by ju5t** [reply]
A better solution: https://github.com/spatie/laravel-livewire-wizard/pull/17. This also supports going back. Navigation was left out on purpose as I think it warrants a different PR.
